### PR TITLE
add detailed log for network requests and responses

### PR DESCRIPTION
**Pull request by amirziarati**
had to load the whole Json to memory to log the response in debug mode. in release mode there is no log and therefore no probable memory problem in production.

### CODE REVIEW

File: traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java [hunk 1]
```diff
@@ -444,11 +444,11 @@ AuthenticatedUrlRequest build(Token token) {
             String tokenValue = (null != token) ? token.getValue() : null;
             return new AuthenticatedUrlRequest(method, getURL(), headers, tokenValue) {
                 @Override
-                public JSONObject getJSONPayload() {
+                public String getPayload() {
```
Comment (ata-n) [inline traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java]: Does this still need to be a String? Can we keep this as a JSONObject?

Comment (amirziarati) [reply traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java]: thsats what we talk about and decided to change to take json out from the router. now all router cares about is the string you want to send as post body. and it does not have to be json.

i was going to put a TODO. but it was easy

File: traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/tasks/NetworkTask.java [hunk 2]
```diff
@@ -1,16 +1,29 @@
 package com.guestlogix.travelercorekit.tasks;
```
[the remark this answers is not in the document]
Comment (amirziarati) [reply traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/tasks/NetworkTask.java]: already did this.

File: traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/utilities/JsonObjectMapper.java [hunk 3]
```diff
@@ -25,4 +36,45 @@ public void onHandleResponse(InputStream stream) {
             callback.onError(new ObjectMappingError(objectMappingFactory, e));
         }
     }
+
+    private InputStream logNetworkResponse(InputStream inputStream) {
```
Comment (OmarPadierna) [inline traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/utilities/JsonObjectMapper.java]: Rename this method into something more related to the class that has it. Meaning, your class JsonMapper is in charge of mapping data to models, as such its log method should be named accordingly.

Comment (amirziarati) [reply traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/utilities/JsonObjectMapper.java]: renemaed it to logRawJson ;)

**Pull request merged**